docs/proposals/20191017-kubeadm-based-control-plane.md

```diff
@@ -454,26 +454,25 @@ spec:
   [etcd best practice](https://etcd.io/docs/v3.3.12/faq/#why-an-odd-number-of-cluster-members).
 - However, allow a control plane using an external etcd cluster to scale up to other numbers such as 2 or 4.
 - Scale up operations must not be done in conjunction with an upgrade operation, this should be enforced using a validation webhook.
-- Scale up operations *could* be blocked based on the Health status
-  - for stacked etcd quorum must be considered
-  - for external etcd only availability of etcd should be considered.
-  - Health status is described below.
+- Scale up operations are blocked based on Etcd and control plane health checks.
+  - See [Healthchecks](#Healthchecks) below.
 
 ![controlplane-init-6](images/controlplane/controlplane-init-6.png)
 
 ##### Scale Down
 
 - Scale down a stacked control plane
-  - For each replica being deleted:
-    - Remove the replica’s etcd member from the etcd cluster
-    - Update the kubeadm generated config map
+  - Wait for deletes to finish before scaling down
+  - Find the oldest machine, or machine marked for removal and target that as the replica to remove
+  - Apply a label to the machine marked for removal (allows for reentrancy)
+  - Remove the replica’s etcd member from the etcd cluster
+  - Update the kubeadm generated config map
+  - Remove the Machine
 - Scale down a control plane with an external etcd
   - Same as above minus etcd management
 - Scale down operations must not be done in conjunction with an upgrade operation, this should not impact manual operations for recovery. This should also be enforced using a validation webhook.
-- Scale down operations *could* be blocked based on the Health status
-  - for stacked etcd quorum must be considered
-  - for external etcd only availability of etcd should be considered.
-  - Health status is described below.
+- Scale down operations are be blocked based on Etcd and control plane health checks
```

Comment:
```suggestion
- Scale down operations are blocked based on Etcd and control plane health checks.
  - If the controller has already picked a machine to scale down, the scale down operation will complete even if that machine's etcd member is not responding. This is required to make scale down re-entrant.
```

Reply:
I shortened that to `  - Etcd health is not required if a machine has already been selected for scale down` and moved the reentrant note to the top along with a note about annotations. Do you feel that captures the essence? I'll need to see the code before i reevaluate if that is not sufficient

Reply:
The current in-tree code does the following:
- Marking a machine for deletion is not blocked on health checking
- Removing the etcd membership for the machine is blocked by an etcd health check failing
- Removing the config map entry is blocked by a control plane components health check failing
- Deletion of the machine is blocked by the etcd membership and config map entry being updated (enforced via annotation added after they have been completed).